### PR TITLE
v2 get data order

### PR DESCRIPTION
**Pull request by Meketh**
### What does this Pull Request do?
Updates to get data order query.

### Are there points in the code the reviewer needs to double check?
Removed deprecated functionality and code refactors. (there will be dragons)
Test will be added on another PR.

### Why was this Pull Request needed?
To be protocol v2 compliant.

### Does this Pull Request meet the acceptance criteria?
- [x] I have read the [Contribution guidelines](https://github.com/wibsonorg/buyer-sdk/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](https://github.com/wibsonorg/buyer-sdk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: buyer-api/src/contractEvents/dataOrderSubscriber.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    'DataOrderCreated',
+    'DataOrderClosed',
+  ],
+  async callback({ returnValues: { orderId: id, owner } }) {
+    const { address } = await getAccount();
+    if (address === owner) {
+      const [storedOrder, chainOrder] = await Promise.all([
+        dataOrders.fetch(id),
+        fetchDataOrder(id),
+      ]);
```
Comment (nicoayala) [inline buyer-api/src/contractEvents/dataOrderSubscriber.js]: Remember that the `orderId` param that comes in the event does not identify a DataOrder in the local db yet, but it identifies the order in the DataExchange. I'd change this do the `fetchDataOrder(id)` first, and then update the local version of the order with the new `id` field.
Also, the order `status` is another field thats needs to be updated to `created`

File: buyer-api/src/services/signingService.js [hunk 2]
```diff
@@ -14,34 +14,34 @@ const url = config.buyerSigningServiceUrl;
...
-const signCloseDataOrder = payload => client.post(
+export const signCloseDataOrder = payload => client.post(
   `${url}/sign/close-data-order`,
   {
     json: payload,
     timeout,
   },
 );
 
-const signinService = {
```
Comment (nicoayala) [inline buyer-api/src/services/signingService.js]: 🤦‍♂️

**Comment by nicoayala** [comment]
I'm merging this PR so I can continue working with these changes. @manelli you can still review it so that @Meketh can include the changes in the following PR

File: buyer-api/src/utils/blockchain.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * @property {Date} createdAt
+ * @property {?Date} closedAt
+ */
+
+/**
+ * @function fetchDataOrder
+ * @param {Number} id Data order id on the data exchange.
+ * @returns {DataOrder} An array of elements stored in the property.
+ */
+export function fetchDataOrder(id) {
```
Comment (nicoayala) [inline buyer-api/src/utils/blockchain.js]: This must be an `async` function

File: buyer-api/src/utils/blockchain.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if (id > 0) {
+    const dataOrder = dx.methods.dataOrders(id).call();
```
Comment (nicoayala) [inline buyer-api/src/utils/blockchain.js]: This call must be `await`ed before `dataOrder` can be used.

File: buyer-api/src/utils/blockchain.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    return {
+      ...dataOrder,
```
Comment (nicoayala) [inline buyer-api/src/utils/blockchain.js]: At this point `dataOrder` is an array so this merge needs to be avoided

**Comment by Meketh** [comment]
Continues on #122

**Pull request merged**